pkg/tracer/tracer.go

```diff
@@ -756,6 +769,65 @@ func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowI
 	return flows
 }
 
+// LookupAndDeleteMap reads all the entries from the eBPF map and removes them from it.
+// TODO: detect whether BatchLookupAndDelete is supported (Kernel>=5.6) and use it selectively
+// Supported Lookup/Delete operations by kernel: https://github.com/iovisor/bcc/blob/master/docs/kernel-versions.md
+func (m *FlowFetcher) LookupAndDeleteMap(met *metrics.Metrics) map[ebpf.BpfFlowId][]ebpf.BpfFlowMetrics {
+	if !m.lookupAndDeleteSupported {
+		return m.legacyLookupAndDeleteMap(met)
+	}
+
+	if !m.lookupAndDeleteIterateSupported {
+		return m.lookupAndDeleteMapUserSpace(met)
+	}
+
+	flowMap := m.objects.AggregatedFlows
+
+	var flows = make(map[ebpf.BpfFlowId][]ebpf.BpfFlowMetrics, m.cacheMaxSize)
+	var metrics = make([]ebpf.BpfFlowMetrics, m.cacheMaxSize*cilium.MustPossibleCPU())
+	var id ebpf.BpfFlowId
+	var ids = make([]ebpf.BpfFlowId, m.cacheMaxSize)
+	var cursor = cilium.MapBatchCursor{}
+
+	count := 0
+	for {
+		count, err := flowMap.BatchLookup(&cursor, ids, metrics, nil)
+		if err == nil || errors.Is(err, cilium.ErrKeyNotExist) {
+			for i, id := range ids[:count] {
+				for j := 0; j < cilium.MustPossibleCPU(); j++ {
+					flows[id] = append(flows[id], metrics[i*cilium.MustPossibleCPU()+j])
+				}
+			}
+
+			break
+		}
+		if err != nil {
+			if errors.Is(err, cilium.ErrNotSupported) {
+				log.WithError(err).Warnf("switching to lookupAndDelete userspace mode")
+				m.lookupAndDeleteIterateSupported = false
+				return m.lookupAndDeleteMapUserSpace(met)
+			}
+			log.WithError(err).WithField("flowId", id).Warnf("couldn't delete flow entry")
+			met.Errors.WithErrorName("flow-fetcher", "CannotDeleteFlows").Inc()
+			continue
+		}
+	}
+
+	// trigger iterator kernel check to look for stale entries
+	f, err := m.flowMapLink.Open()
+	if err != nil {
+		log.WithError(err).WithField("flowId", id).Warnf("couldn't open flows map iterator")
+		m.lookupAndDeleteIterateSupported = false
+		return m.lookupAndDeleteMapUserSpace(met)
+	}
+	f.Close()
```

Comment:
It's not enough to open and close the link, you also need to read from it. See: 

https://github.com/xdp-project/bpf-examples/blob/master/pping/pping.c#L696

Reply:
ok I know I was missing something with iterator userspace trigger